docs/tutorials/semantic_search/asymmetric_embedding_model.md

```diff
@@ -0,0 +1,266 @@
+# Tutorial: Generating Embeddings Using a Local Asymmetric Embedding Model in OpenSearch
+
+This tutorial demonstrates how to generate text embeddings using an asymmetric embedding model in OpenSearch, implemented within a Docker container. The example model used in this tutorial is the multilingual `intfloat/multilingual-e5-small` model from Hugging Face. You will learn how to prepare the model, register it in OpenSearch, and run inference to generate embeddings.
+
+> **Note**: Make sure to replace all placeholders (e.g., `your_`) with your specific values.
+
+---
+
+## Prerequisites
+
+- Docker Desktop installed and running on your local machine.
+- Basic familiarity with Docker and OpenSearch.
+- Access to the Hugging Face model `intfloat/multilingual-e5-small` (or another model of your choice).
+---
+
+## Step 1: Spin Up a Docker OpenSearch Cluster
+
+To run OpenSearch in a local development environment, you can use Docker and a pre-configured `docker-compose` file.
+
+### a. Update Cluster Settings
+
+Before proceeding, ensure your cluster is configured to allow registering models. You can do this by updating the cluster settings via the following request:
+
+```
+PUT _cluster/settings
+{
+  "persistent": {
+    "plugins.ml_commons.allow_registering_model_via_url": "true",
+    "plugins.ml_commons.only_run_on_ml_node": "false",
+    "plugins.ml_commons.model_access_control_enabled": "true",
+    "plugins.ml_commons.native_memory_threshold": "99"
+  }
+}
+```
+
+This configuration ensures that OpenSearch can accept machine learning models from external URLs and can run models across non-ML nodes.
+
+### b. Use a Docker Compose File
+
+You can use this sample [file](https://opensearch.org/docs/latest/install-and-configure/install-opensearch/docker/#sample-docker-compose-file-for-development) as an example.
+Once your `docker-compose.yml` file is ready, run the following command to start OpenSearch in the background:
+
+```
+docker-compose up -d
+```
+
+---
+
+## Step 2: Prepare the Model for OpenSearch
+
+In this tutorial, we’ll use the Hugging Face model `intfloat/multilingual-e5-small`, which is capable of generating multilingual embeddings. Follow these steps to prepare and zip the model for use in OpenSearch.
+
+### a. Clone the Model from Hugging Face
+
+To download the model, use the following steps:
+
+1. Install Git Large File Storage (LFS) if you haven’t already:
+
+   ```
+   git lfs install
+   ```
+
+2. Clone the model repository:
+
+   ```
+   git clone https://huggingface.co/intfloat/multilingual-e5-small
+   ```
+
+This will download the model files into a directory on your local machine.
+
+### b. Zip the Model Files
+
+In order to upload the model to OpenSearch, you must zip the necessary model files (`model.onnx`, `sentencepiece.bpe.model`, and `tokenizer.json`). The `model.onnx` file is located in the `onnx` directory of the cloned repository.
```

Comment:
Is the onnx the only format that this model provided? Can we add pytorch format tutorial here?

Reply:
Hey Zane! I can see why you ask Ill clarify that this is only for onnx, I havent used pytorch models so thats I wrote it like that

Reply:
Got it, my suggestion is we can add both onnx and pytorch cases so that user can choose between them based on their cases.